### PR TITLE
download tests

### PR DESCRIPTION
**Pull request by ralphtheninja**
So after some poking around I ended up with this. I found a couple of bugs. See comments below.

### CODE REVIEW

**Comment by ralphtheninja** [comment]
@mafintosh Maybe after some adjustments we could squash it once we're content?

**Comment by mafintosh** [comment]
:+1: for squashing

File: download.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    pump(fs.createReadStream(cachedPrebuild), zlib.createGunzip(), tfs.extract(rc.cwd, {readable: true, writable: true}).on('entry', updateName), function (err) {
+      if (err) return cb(err)
+      if (binaryName) {
+        try {
+          var resolved = path.resolve(rc.cwd, binaryName)
+          require(resolved)
+          log.info('unpack', 'required ' + resolved + ' successfully')
+          return cb()
```
Comment (mafintosh) [inline download.js]: don't try-catch the cb() call. we might end up calling cb twice when doing this

File: test/util-test.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
+var test = require('tape')
+var util = require('../util')
+var fs = require('fs')
+
+test('prebuildCache() ends with _prebuilds folder', function (t) {
+  process.env.APPDATA = 'somepathhere'
+  t.ok(util.prebuildCache().indexOf('/npm-cache/_prebuilds') > 0, 'APPDATA set')
```
Comment (mafintosh) [inline test/util-test.js]: shouldn't this check if it contains APPDATA?

File: util.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  })
+}
+
+function expandTemplate (template, values) {
+  Object.keys(values).forEach(function (key) {
+    var regexp = new RegExp('\{' + key + '\}', 'g')
+    template = template.replace(regexp, values[key])
+  })
+  return template
+}
```
Comment (mafintosh) [inline util.js]: you should put this one in a module

File: download.js (hunk 1 above)
Comment (ralphtheninja) [inline download.js]: Good catch!

File: test/util-test.js (hunk 2 above)
Comment (ralphtheninja) [inline test/util-test.js]: Yes it should! :)

File: util.js (hunk 3 above)
Comment (ralphtheninja) [inline util.js]: Good idea. I'll do that.

File: download.js (hunk 1 above)
Comment (ralphtheninja) [inline download.js]: Updated.

File: test/util-test.js (hunk 2 above)
Comment (ralphtheninja) [inline test/util-test.js]: Updated.

**Comment by ralphtheninja** [comment]
@mafintosh Ready for squashing?

**Comment by mafintosh** [comment]
@ralphtheninja looks good! could you add a test script to npm as well? then i'll add travis afterwards

**Comment by mafintosh** [comment]
lol script is already there. ready for squashing :tada:

**Comment by mafintosh** [comment]
2.3.2

**Pull request merged**